### PR TITLE
Add script to regenerate all codecs

### PR DESCRIPTION
**Pull request by TimvdLippe**
The `build-all.sh` script runs `npm run build` for all codecs.
Additionally, recompile all codecs using the script with the current
state of the repository. This is in preparation of making some build
changes for #1152

### CODE REVIEW

File: codecs/build-all.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CURRENT_DIRECTORY=$(dirname "$(readlink -f "$0")")
+
+for codec in "avif" "hqx" "imagequant" "jxl" "mozjpeg" "oxipng" "png" "resize" "rotate" "visdif" "webp" "wp2"
```
Comment (surma) [inline codecs/build-all.sh]: Any chance we can just go into each directory? If someone adds a codec, they’ll 100% forget to add it to this list.

**Review by surma — COMMENTED** [review]
Bit surprised that the Wasm binaries changed. They should have been identical as far as I can tell. Any idea why?

also cc @RReverser

File: codecs/build-all.sh (hunk 1 above)
Comment (TimvdLippe) [reply codecs/build-all.sh]: Yup, turns out we can do that :smile:

**Comment by TimvdLippe** [comment]
Unfortunately, I don't know why they have changed. There is also the extra `.d.ts` file which was missing before maybe?

I did build it locally and ran through all codecs to check if they were still working, which they were. So I suppose it is fine? :sweat_smile:

**Comment by TimvdLippe** [comment]
It seems like the MakeFiles don't recompile sometimes. This means that you manually have to delete files to force a rebuild, which then shows the new result. Maybe we should force clean the output files whenever we build, to ensure we always build fresh?

**Comment by surma** [comment]
The whole point of Makefiles is to only rebuild files that have changed. We do have `make clean` in all Makefiles to force a rebuild tho. I am not sure we should do that _by default_ from the script, but a parameter/env to force a rebuild seems sensible.

**Comment by TimvdLippe** [comment]
I did attempt running `make clean`, but that in itself was throwing errors complaining about missing files or something. I would have expected Make to figure out that the environment variables had changed, but maybe these environment variables are missing? E.g. make detects that the scripts nor the files have changed, so it bails out. But it accidentally overlooks the fact that these environment variables can affect the build?

**Pull request closed without merging**